File: Flip54/Design/DS.swift
import SwiftUI

enum DS {
    enum Colors {
        static let bg            = Color(hex: "#111111")
        static let bgRaised      = Color(hex: "#191919")
        static let bgCard        = Color(hex: "#1A1A1A")
        static let surface       = Color(hex: "#101010")
        static let border        = Color(hex: "#2A2A2A")
        static let borderSub     = Color(hex: "#1E1E1E")
        static let textPrimary   = Color(hex: "#FEFEFE")
        static let textSecondary = Color(hex: "#D1CDC9")
        static let textTertiary  = Color(hex: "#666460")
        static let cardFace      = Color(hex: "#FEFEFE")
        static let red           = Color(hex: "#C8262C")
        static let redSoft       = Color(hex: "#2A1214")
        static let gold          = Color(hex: "#D1C4B1")
        static let goldSoft      = Color(hex: "#1E1B16")
        static let goldLight     = Color(hex: "#F0EAE0")
        static let neutral       = Color(hex: "#D1CDC9")
        static let success       = Color(hex: "#4CAF6E")
        static let urgent        = Color(hex: "#E8543C")
        static let white         = Color(hex: "#FEFEFE")
    }

    enum Typography {
        // Display — Barlow Condensed ExtraBold (maps to Futura Extra Bold Condensed in spec)
        static func display(_ text: String, size: CGFloat = 50) -> some View {
            Text(text)
                .font(.custom("BarlowCondensed-ExtraBold", size: size))
        }
        // Subheading — Oswald SemiBold (maps to Trade Gothic in spec)
        static func sub(_ text: String, size: CGFloat = 34) -> some View {
            Text(text)
                .font(.custom("Oswald-SemiBold", size: size))
        }
        // Body — system Helvetica
        static func body(_ text: String, size: CGFloat = 18) -> some View {
            Text(text)
                .font(.system(size: size, weight: .regular, design: .default))
        }
        // Monospace — IBM Plex Mono
        static func mono(_ text: String, size: CGFloat = 14) -> some View {
            Text(text)
                .font(.custom("IBMPlexMono-Medium", size: size))
        }
        // Caption
        static func caption(_ text: String, size: CGFloat = 13) -> some View {
            Text(text)
                .font(.system(size: size, weight: .regular))
        }
    }

    enum Layout {
        static let horizontalMargin: CGFloat = 24
        static let cornerRadius: CGFloat = 14
        static let buttonHeight: CGFloat = 56
        static let cardWidth: CGFloat = 280
        static let cardHeight: CGFloat = 392
        static let cardCornerRadius: CGFloat = 14
    }

    enum Shadow {
        static let card = Color.black.opacity(0.7)
    }
}

extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let r = Double((int >> 16) & 0xFF) / 255
        let g = Double((int >> 8) & 0xFF) / 255
        let b = Double(int & 0xFF) / 255
        self.init(red: r, green: g, blue: b)
    }
}

File: android/app/src/main/kotlin/com/flip54/app/ui/theme/DS.kt
package com.flip54.app.ui.theme

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.flip54.app.R

object DS {

    object Colors {
        val bg            = Color(0xFF111111)
        val bgRaised      = Color(0xFF191919)
        val bgCard        = Color(0xFF1A1A1A)
        val surface       = Color(0xFF101010)
        val border        = Color(0xFF2A2A2A)
        val borderSub     = Color(0xFF1E1E1E)
        val textPrimary   = Color(0xFFFEFEFE)
        val textSecondary = Color(0xFFD1CDC9)
        val textTertiary  = Color(0xFF666460)
        val cardFace      = Color(0xFFFEFEFE)
        val red           = Color(0xFFC8262C)
        val redSoft       = Color(0xFF2A1214)
        val gold          = Color(0xFFD1C4B1)
        val goldSoft      = Color(0xFF1E1B16)
        val goldLight     = Color(0xFFF0EAE0)
        val neutral       = Color(0xFFD1CDC9)
        val success       = Color(0xFF4CAF6E)
        val urgent        = Color(0xFFE8543C)
        val white         = Color(0xFFFEFEFE)
    }

    object Fonts {
        val barlow    = FontFamily(Font(R.font.barlow_condensed_extrabold, FontWeight.ExtraBold))
        val oswald    = FontFamily(Font(R.font.oswald_semibold, FontWeight.SemiBold))
        val ibmMono   = FontFamily(Font(R.font.ibm_plex_mono_medium, FontWeight.Medium))
    }

    object Type {
        fun display(size: Float = 50f) = TextStyle(
            fontFamily = Fonts.barlow,
            fontWeight = FontWeight.ExtraBold,
            fontSize = size.sp,
            color = Colors.textPrimary
        )
        fun sub(size: Float = 34f) = TextStyle(
            fontFamily = Fonts.oswald,
            fontWeight = FontWeight.SemiBold,
            fontSize = size.sp,
            color = Colors.textPrimary
        )
        fun mono(size: Float = 14f) = TextStyle(
            fontFamily = Fonts.ibmMono,
            fontWeight = FontWeight.Medium,
            fontSize = size.sp,
            color = Colors.textPrimary
        )
    }

    object Layout {
        val horizontalMargin: Dp = 24.dp
        val cornerRadius: Dp = 14.dp
        val buttonHeight: Dp = 56.dp
        val cardWidth: Dp = 280.dp
        val cardHeight: Dp = 392.dp
        val cardCornerRadius: Dp = 14.dp
    }
}
